Default-implemented broadcasting operators such as `+`, `-` and `*` heavily impacted type checker performance and caused the TensorFlow library to fail to compile.
```console
error: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions
    let loss = maxLogitsWithZero - logits * labels + log(1 + exp(-abs(logits)))
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

So I removed all operators in `VectorProtocol` extensions.

--- a/test/AutoDiff/refcounting.swift
+++ b/test/AutoDiff/refcounting.swift
@@ -23,6 +23,8 @@ public struct Vector : AdditiveArithmetic, VectorProtocol, Differentiable, Equat
   @differentiable(vjp: fakeVJP)
   public static func - (lhs: Vector, rhs: Vector) -> Vector { abort() }
 
+  public func adding(_ scalar: Float) -> Vector { abort() }
+  public func subtracting(_ scalar: Float) -> Vector { abort() }
   public func scaled(by scalar: Float) -> Vector { abort() }
 
   public static func fakeVJP(lhs: Vector, rhs: Vector) -> (Vector, (Vector) -> (Vector, Vector)) { abort() }

--- a/test/Sema/struct_key_path_iterable.swift
+++ b/test/Sema/struct_key_path_iterable.swift
@@ -108,8 +108,8 @@ struct DummyOptimizer<P : KeyPathIterable, Scalar : BinaryFloatingPoint>
     parameters: inout P, withGradients gradients: P
   ) {
     for kp in parameters.recursivelyAllWritableKeyPaths(to: Tensor<Scalar>.self) {
-      firstMoments[keyPath: kp] *= learningRate
-      parameters[keyPath: kp] -= learningRate * parameters[keyPath: kp]
+      firstMoments[keyPath: kp].scale(by: learningRate)
+      parameters[keyPath: kp] -= parameters[keyPath: kp].scaled(by: learningRate)
     }
   }
 }

--- a/test/Sema/struct_vector_protocol.swift
+++ b/test/Sema/struct_vector_protocol.swift
@@ -7,16 +7,28 @@ _ x: inout T, scalar: T.VectorSpaceScalar
   // Test `AdditiveArithmetic` requirements: `zero`, `+`, `-`.
   let zero = T.zero
   x += x + zero
-  x -= x - zero
+  x += x - zero
+  // Test `VectorProtocol` requirements: `VectorSpaceScalar`, `adding(_:)`, `add(_:)`
+  // `subtracting(_:)`, `subtract(_:)`, `scaled(by:)`, and `scale(by:)`.
+  x.add(scalar)
+  x.add(scalar)
+  x.scale(by: scalar)
+  _ = x.adding(scalar)
+  _ = x.subtracting(scalar)
+  _ = x.scaled(by: scalar)
+   
+  // NOTE: Operators have been disabled for type checker performance reasons.
+  // x += x + zero
+  // x -= x - zero
   // Test `VectorProtocol` requirements: `VectorSpaceScalar`, `+`, `-`, `*`.
-  x += scalar
-  x -= scalar
-  x *= scalar
-  _ = x + scalar
-  _ = scalar + x
-  _ = x - scalar
-  _ = scalar * x
-  _ = x * scalar
+  // x += scalar
+  // x -= scalar
+  // x *= scalar
+  // _ = x + scalar
+  // _ = scalar + x
+  // _ = x - scalar
+  // _ = scalar * x
+  // _ = x * scalar
 }
 
 struct Float2: VectorProtocol {

--- a/utils/update_checkout/update-checkout-config.json
+++ b/utils/update_checkout/update-checkout-config.json
@@ -375,7 +375,7 @@
                 "clang-tools-extra": "swift-DEVELOPMENT-SNAPSHOT-2019-06-06-a",
                 "libcxx": "swift-DEVELOPMENT-SNAPSHOT-2019-06-06-a",
                 "tensorflow": "ebc41609e27dcf0998d8970e77a2e1f53e13ac86",
-                "tensorflow-swift-apis": "5d3ef57b501781f1bab3b4ca85e8b8fc91671c14",
+                "tensorflow-swift-apis": "7a3ed481bba53a7cd82f8a46c0df9f09a6e9747f",
                 "indexstore-db": "swift-DEVELOPMENT-SNAPSHOT-2019-06-06-a",
                 "sourcekit-lsp": "swift-DEVELOPMENT-SNAPSHOT-2019-06-06-a"
             }

